contracts/DependencyRegistryV0.sol

```diff
@@ -0,0 +1,923 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity 0.8.17;
+
+// Created By: Art Blocks Inc.
+
+import "./interfaces/0.8.x/IAdminACLV0.sol";
+import "./interfaces/0.8.x/IDependencyRegistryCompatibleV0.sol";
+import "./interfaces/0.8.x/IDependencyRegistryV0.sol";
+
+import "@openzeppelin-4.7/contracts/utils/Strings.sol";
+import "@openzeppelin-4.7/contracts/utils/structs/EnumerableSet.sol";
+import "@openzeppelin-4.8/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "@openzeppelin-4.8/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin-4.5/contracts/utils/math/SafeCast.sol";
+
+import "./libs/0.8.x/BytecodeStorage.sol";
+import "./libs/0.8.x/Bytes32Strings.sol";
+
+/**
+ * @title Art Blocks Dependency Registry, V0.
+ * @author Art Blocks Inc.
+ * @notice Privileged Roles and Ownership:
+ * Permissions managed by ACL contract. If/when we ever call
+ * renounceOwnership() this will becom a frozen, immutable registry
+ * as no upgrades will be possible.
+ * This contract has been made upgradeable as we expect
+ * its required functionality in relation to the Art Blocks ecosystem
+ * to evolve over time.
```

Comment:
Would add onto this sentence at the beginning: "This contract is intended to be an auxiliary reference registry to our non-upgradeable and immutable ERC-721 conforming core contracts, and has been made upgradeable..."